dvc/repo/fetch.py

```diff
@@ -22,7 +23,18 @@ def fetch(
             remote=remote,
             jobs=jobs,
             recursive=recursive,
-        )["local"]
-        return self.cloud.pull(
-            used, jobs, remote=remote, show_checksums=show_checksums
         )
+
+        for dep in used["repo"]:
+            dep.fetch()
+
+        try:
+            return self.cloud.pull(
+                used["local"],
+                jobs,
+                remote=remote,
+                show_checksums=show_checksums,
+            ) + len(used["repo"])
+        except ConfigError:
+            if not used["repo"]:
```

Comment:
You are trying to make sure that `dvc fetch` works when there is no remote setup but there are repo imports? In that case this `except ConfigError` is too general, since it covers all config errors and results in incorrect behaviour.

Btw, if dep.fetch fails, `dvc fetch` won't download anything from remote here, which is wrong. We should fetch everything we can and then error-out.

Reply:
> You are trying to make sure that dvc fetch works when there is no remote setup but there are repo imports?

Yes

> In that case this except ConfigError is too general, since it covers all config errors and results in incorrect behaviour.

Know :disappointed:  we can define a separate exception for this specific case, or leave it as it was before (a guard clause under `get_remote`)

https://github.com/iterative/dvc/blob/master/dvc/data_cloud.py#L63-L68

> Btw, if dep.fetch fails, dvc fetch won't download anything from remote here, which is wrong. We should fetch everything we can and then error-out.

:thinking: I guess I'll `dep.fetch` after `cloud.pull`

Reply:
This still looks wrong. If remote is not configured and there are imports than the error will be silently ignored.

Reply:
@mroutis This one is not resolved, @Suor is right, we shouldn't ignore this error silently if we have something to pull. But I guess we could ignore this one if there are no non-imported outputs to pull.

Reply:
@efiop , that's right, it should be `... and not used["local"]`